cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -460,9 +469,73 @@ void Refactor::CollapseNode(mgp_list *args, mgp_graph *memgraph_graph, mgp_resul
           return;
         }
       }
-    } 
-      
-    
+    }
+
+  } catch (const std::exception &e) {
+    record_factory.SetErrorMessage(e.what());
+    return;
+  }
+}
+
+void Refactor::ExtractNode(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    mgp::Graph graph{memgraph_graph};
+    mgp::Value rel_or_id{arguments[0]};
+    auto labels{arguments[1].ValueList()};
+    auto out_type{arguments[2].ValueString()};
+    auto in_type{arguments[3].ValueString()};
+
+    auto extract = [&](mgp::Relationship relationship) {
+      auto new_node = graph.CreateNode();
+      for (const auto &label : labels) {
```

Comment:
I think you don't need here reference, since labels is returning string_view and it is not copied actually

Reply:
labels is a list which contains mgp::Value, not string_view

Reply:
Ah, sorry, I got something wrong from the code, thought that we got labels from `node.Labels()`. All good!